### ISSUE TITLE
BreakMeshByBlock connectivity 

### ISSUE THREAD

**Issue by hugary1995** [idaholab/moose#21161]
This is a more specific issue for the second bullet point in #12033.

Quote the original issue:
> the connectivity of the elements with news nodes are not correct -> lindsayad comment: I don't believe we can ever get this correct since in a strict sense after node replacement the former neighbors are not neighbors anymore

@lindsayad said we can't get this correct...so, we should probably think of another way of handling this. Because we'd really want this kind of mesh generator to work with anything that calls meshChanged()

**Comment by hugary1995** [comment]
I guess one way for czm to work is, instead of abusing the usage of BreakMeshByBlockGenerator, to explicitly add lowerD elements between broken elements.

**Comment by lindsayad** [comment]
But interface kernels require the higher dimensional elements to share a side. I think the long-term solution for CZM, to maximize its robustness, is probably to use/create a system that doesn't expect "neighbors" to be libMesh neighbors. There is of course mortar and other constraints, although I think the argument against mortar is that it's too heavy for the conceptual need. @arovinelli always told me that even if the mesh displaces, he always wants the CZM "neighbors" to be the original libMesh neighbors

**Comment by arovinelli** [comment]
@jiangwen84 and @lindsayad traditional CZM assumes neighbors are fixed and connected by an additional element between the cohesive surfaces. The model we have right now captures large sliding using a total Lagrangian formulation but always assuming neighbors are fixed. An idea to merge the two models might be to make `CZMMaterials` available to  mortar or other surface-to-surface constraints, so that we don't confuse users with the meaning of CZM.

**Comment by hugary1995** [comment]
But at heart of czm, even traditional czm, all we need to know is the displacement jump across the interface. If the mortar in moose can track sliding, that's really all we need, then I won't care anymore whether the neighbor pairs are fixed.

**Comment by hugary1995** [comment]
in other words, I view "fixed neighbors" as an implementational choice, and it doesn't have to be tied to "traditional" czm.

**Comment by lindsayad** [comment]
I would be happy to help, time allowing, with creating a CZM version based off of our mortar system. Perhaps @recuero might be able to assist as well

**Comment by arovinelli** [comment]
@hugary1995 I think CZM as is works for most applications and users, and definitely have computational advantages compared to mortar. I agree a mortar implementation will be more flexible, but it would serve only a few special cases --@jiangwen84 probably has something in mind--.
To be back  on topic, @lindsayad once proposed to add `fakeneighbors`in libmesh and make the CZM use them instead of `topologicalnbeighbors` I guess this might be the best approach to ensure `breakmeshbyblock` works properly

**Comment by hugary1995** [comment]
@roystgnr The difference is that an element cannot figure out its "fake" neighbor without additional data structures. Let me make the definitions here for the benefit of following discussion:
- Topological neighbors are a pair of elements that share a node(1D)/edge(2D)/face(3D). Hence it is trivial to figure out an element's topological neighbors at any time from the node-to-element map.
- Using the above definition, fake neighbors are a pair of elements that were once topological neighbors but no longer are, e.g. they broke up.

In short, "topological neighbors vs fake neighbors" is like "neighbors vs ex-neighbors". So if we expand the API, the first thing we might want to do is to store an element's ex-neighbor when they break up.

**Comment by lindsayad** [comment]
I'm going to re-open this since we've (temporarily) reverted the code that addresses this
